cli/medperf/comms/entity_resources/resources.py

```diff
@@ -14,9 +14,46 @@
 
 import shutil
 import os
+import logging
+import yaml
 import medperf.config as config
 from medperf.utils import generate_tmp_path, get_cube_image_name, remove_path, untar
-from .utils import download_resource
+from .utils import download_resource, verify_or_get_hash
+
+
+def _should_get_cube(output_path, expected_hash):
+    if os.path.exists(output_path) and expected_hash:
+        if verify_or_get_hash(output_path, expected_hash):
+            logging.debug(f"{output_path} exists and is up to date")
+            return False
+        logging.debug(f"{output_path} exists but is out of date")
+    return True
+
+
+def _should_get_cube_params(output_path, expected_hash):
+    # Happens to be the same logic
+    return _should_get_cube(output_path, expected_hash)
```

Comment:
Might be pedantic, but in this case we should have a generic name for this function (`should_get_file` or `should_get_asset`?) and have both `_should_get_cube` and `_should_get_cube_params` call that. It makes it just a bit more readable.